lib/perl/Genome/Config/AnalysisProject/Command/ReplaceModel.pm

```diff
@@ -0,0 +1,145 @@
+       new_model => { is => 'Genome::Model', },
+   },
+};
+
+sub help_brief {
+    return 'replace a model';
+}
+
+sub help_detail {
+    return <<EOS
+This command takes a model and new config profile item and copies the model, replacing the different items in the new profile config as compared to the model's existing config. The model is queued for building. Additionally, the builds forthe old model shold be abandoned. That way they do not appear in a failed state.
+
+Use this command when a model needs to replaced on a current analysis project. This is usually due to a need to switch a model parameter because the current configuration fails to produce a successful build. If a different config is needed for most or all instrument data, a new config should be added and the instrument data reprocessed.
+
+The new config profile must have been previously added to the analysis project. Additionally, the profile status must be 'inactive'.
+
+See analysis project configuration documentation for more details on creating/finding a new config item.
+
+EOS
+}
+
+sub execute {
+    my $self = shift;
+    $self->status_message('Replace models on analysis project...');
+
+    $self->_verify_new_config;
+    $self->status_message('New config: %s', $self->new_profile_item->file_path);
+    $self->status_message('Old model:  %s', $self->model->__display_name__);
+    my $overrides = $self->_overrides_for_model;
+    $self->_copy_model($overrides);
+
+    $self->status_message('NEW model:  %s', $self->new_model->__display_name__);
+    $self->status_message('Please abandon builds for old model!');
+    return 1;
+}
+
+sub _verify_new_config {
+    my $self = shift;
+
+    if ( $self->new_profile_item->status ne 'inactive') {
+        die $self->error_message(
+            "Config profile (%s) status (%s) must be 'inactive'.", $self->new_profile_item->id, $self->new_profile_item->status
+        );
+    }
+
+    if ( not $self->analysis_project->is_current ) {
+        die $self->error_message(
+            "Analysis project (%s) status (%s) is not a 'current' status.", $self->analysis_project->id, $self->analysis_project->status
+        );
+    }
+
+    return 1;
+}
+
+sub _overrides_for_model {
+    my $self = shift;
+
+    my $config_for_model = $self->_load_model_config_from_file( $self->model->config_profile_item->file_path );
+    my $config_for_new_model = $self->_load_model_config_from_file( $self->new_profile_item->file_path );
+    my %overrides;
+    for my $key ( keys %$config_for_model ) {
+        next if ref $config_for_model->{$key};
+        next if defined $config_for_new_model->{$key} and $config_for_model->{$key} eq $config_for_new_model->{$key};
+        my $key_no_id = $key;
+        $key_no_id =~ s/_id//;
+        $overrides{$key_no_id} = $config_for_new_model->{$key};
+    }
+
+    if ( not %overrides ) {
+        die $self->error_message("No overrides found for model! %s\nConfig: %s\nNew Config: %s\n",
```

Comment:
We have `$self->fatal_message` now to achieve the effect of `die $self->error_message`.